spec/javascripts/govuk_publishing_components/analytics-ga4/ga4-event-tracker.spec.js

```diff
@@ -412,4 +412,65 @@ describe('Google Analytics event tracking', function () {
       expect(window.dataLayer[0].event_data.url).toEqual(null)
     })
   })
+
+  describe('doing tracking on the feedback component', function () {
+    beforeEach(function () {
+      var attributes = {
+        event_name: 'form_submit',
+        type: 'feedback',
+        text: 'Yes',
+        section: 'Is this page useful?'
+      }
+      element.innerHTML =
+        '<div data-module="ga4-event-tracker">' +
+      '</div>'
+
+      var yesButton = document.createElement('button')
+      yesButton.setAttribute('data-ga4', JSON.stringify(attributes))
+      yesButton.id = 'yesButton'
+
+      attributes.text = 'No'
+      var noButton = document.createElement('button')
+      noButton.setAttribute('data-ga4', JSON.stringify(attributes))
+      noButton.id = 'noButton'
+
+      element.appendChild(yesButton)
+      element.appendChild(noButton)
+
+      document.body.appendChild(element)
+      new GOVUK.Modules.Ga4EventTracker(element).init()
+
+      window.dataLayer = []
+    })
+
+    it('should track "Yes" button clicks', function () {
+      expected = new GOVUK.analyticsGA4.Schemas().eventSchema()
+
+      expected.event = 'event_data'
```

Comment:
Could `expected` be set in the `beforeEach` and then just set `expected.event_data.text` in the individual tests? Only because it looks like all the properties are the same except for `expected.event_data.text`. Although if you feel that makes it more confusing to read, happy to leave it as it is 👍

Reply:
Thanks @JamesCGDS - I've changed this to reflect what you've said :+1: